### ISSUE TITLE
A quick question on constructing "L" object

### ISSUE THREAD

**Issue by gabriellajg** [csoneson/conquer_comparison#1]
This is amazing work!! I really appreciate that you made the expression data and the source codes available. 

I was trying to apply your function on my data and was having a hard time constructing an "L" object to feed into any apply*.R function. Say, I would like to apply run_BPSC() on data set GSE74596, so I did the following things:

GSE74596 <- readRDS("GSE74596.rds")
GSE74596 <- updateObject(GSE74596)
experiments(GSE74596)
(GSE74596_gene <- experiments(GSE74596)[["gene"]])
L <- list()
L$count <- assays(GSE74596_gene)[["count"]]
L$condt <- colData(GSE74596)$characteristics_ch1.5

Could you please let me know if this is the right way to prepare an "L" object or if I need another function to do it? Any comments would be greatly appreciated. 

Please let me know if this is not the right place to post questions, 

Thanks,

Gabby

**Comment by csoneson** [comment]
Hi Gabby,

the functions in the apply_* scripts are called from the `run_diffexpression.R` script, which also does all the preprocessing and prepares the list L from the original MultiAssayExperiment object. In the process, it calls the cleaning and subsetting functions in the `prepare_mae.R` script, to remove rows corresponding to ERCC spike-ins, subset to predefined groups and filter the expression matrix. In principle, I think what you have above should be enough as a minimal L list, except that you may need L$condt to be a named vector, with names matching the column names of L$count. Could you let me know what type of problem you are having? Also note that the code is adapted and in some places limited to two-group comparisons, since that was the focus of our study. 

Charlotte

**Comment by gabriellajg** [comment]
Hi Charlotte,

Thanks for your response. I also want to clean my data set before feeding them into the functions for differential analysis, and I was having some problems creating objects like args, config_file, and config as used in your run_diffexpression.R file I am wondering if you can give me a quick walk through of the process?  Say, I have the object L I created in my first post, how could I clean the data and feed them into run_SeuratBimod() function?

Thank you,

Gabby

**Comment by csoneson** [comment]
Hi Gabby,

the "args" lines is just there since I call the R scripts from the command line via the makefile, and I need a way to provide arguments to the code. 
The "config_file" is a configuration file for each data set (located in the "config" folder), which lists which MultiAssayExperiment object to use, which groups to compare, the sample sizes and number of repeated subsamplings, where to write output etc. 

To apply the cleaning functions, you need to have your data in a MultiAssayExperiment object, like the ones that we used in our comparison. The clean_mae() function basically just removes ERCC spike-ins, so if you don't have them you don't need to do that. 

The subset_mae() function first extracts a pre-defined collection of samples from the data set and defines a named vector with the grouping information. If you already have that information, you don't need to do that either. Then it does some filtering of genes with low expression (lines 40-61 of prepare_mae.R). This you can just as well do directly on the count matrix. 

So to summarise, if you want to apply our functions, you need to have your data in a MultiAssayExperiment object, and you need to tell the function which samples to retain and what group they belong to (for our comparisons, we generated this information with the generate_subsets.R script). However, if you already have the count matrix and the group vector that you want to use, you can just remove the ERCC spike-ins (if applicable) and filter the matrix manually before providing the object to run_SeuratBimod(). 

Charlotte
